modules/entity_version_history/src/Entity/HistoryTabSettings.php

```diff
@@ -0,0 +1,204 @@
+  /**
+   * The target field (machine name).
+   *
+   * @var string
+   */
+  protected $target_field = '';
+
+  /**
+   * Constructs a HistoryTabSettings object.
+   *
+   * @param array $values
+   *   An array of the referring entity bundle with:
+   *   - target_entity_type_id: The entity type.
+   *   - target_bundle: The bundle.
+   *   Other array elements will be used to set the corresponding properties on
+   *   the class; see the class property documentation for details.
+   * @param string $entity_type
+   *   The entity type id.
+   */
+  public function __construct(array $values, string $entity_type = 'entity_version_history_settings') {
+    if (empty($values['target_entity_type_id'])) {
+      throw new HistoryTabSettingsException('Attempt to create entity version history settings without a target_entity_type_id.');
+    }
+    if (empty($values['target_bundle'])) {
+      throw new HistoryTabSettingsException('Attempt to create entity version history settings without a target_bundle.');
+    }
+    parent::__construct($values, $entity_type);
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.ShortMethodName)
+   */
+  public function id(): string {
+    return $this->target_entity_type_id . '.' . $this->target_bundle;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetEntityTypeId(): string {
+    return $this->target_entity_type_id;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetBundle(): string {
+    return $this->target_bundle;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getTargetField(): string {
+    return $this->target_field;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setTargetBundle(string $target_bundle): ConfigEntityInterface {
+    $this->target_bundle = $target_bundle;
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setTargetField(string $target_field): ConfigEntityInterface {
+    $this->target_field = $target_field;
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preSave(EntityStorageInterface $storage): void {
+    $this->id = $this->id();
+    parent::preSave($storage);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function calculateDependencies(): ConfigEntityInterface {
+    parent::calculateDependencies();
+
+    // Create dependency on the bundle.
+    $entity_type = \Drupal::entityTypeManager()->getDefinition($this->target_entity_type_id);
+    $bundle_config_dependency = $entity_type->getBundleConfigDependency($this->target_bundle);
+
+    if (!empty($this->target_field)) {
+      $field_config = \Drupal::entityTypeManager()->getStorage('field_config')->load("$this->target_entity_type_id.$this->target_bundle.$this->target_field");
+      $this->addDependency($field_config->getConfigDependencyKey(), $field_config->getConfigDependencyName());
+    }
+
+    $this->addDependency($bundle_config_dependency['type'], $bundle_config_dependency['name']);
```

Comment:
Why is this being added after the field one? Move it up with the block on the bundle config dependency.